src/services/character/character.service.ts

```diff
@@ -0,0 +1,30 @@
+import _ from 'lodash'
+import { Query } from 'mongoose'
+import characters from '../../data/models/character-schema'
+import { CharacterType } from '../../types/character/character-type'
+
+export class CharacterService {
+  private static _instance: CharacterService
+
+  public static getInstance(): CharacterService {
+    if(_.isNil(CharacterService._instance)) {
+      CharacterService._instance = new CharacterService()
+    }
+    return CharacterService._instance
+  }
+
+  public async getEntity(id: string) {
+    return await characters.findOne({ ownerId: id }, (err: Error, characterFound) => {        
+      if (characterFound) return characterFound
+      else throw err
+    })
+  }
+
+  public setEntity(id: string, characterToInsert: CharacterType) {
```

Comment:
You should start adding return types to ALL your methods.
You will force your code to follow the safe guard type about what it should return and it will help a lot to spot the errors during developtment.

Reply:
Will do... ><